### PR TITLE
Captions component in composite

### PR DESCRIPTION
**Pull request by carocao-msft**
# What
Captions component in call and callwithchat composite

# Why

# How Tested

Uploading demo.mov…

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/src/composites/CallComposite/components/CallArrangement.tsx [hunk 1]
```diff
@@ -154,6 +156,8 @@ export const CallArrangement = (props: CallArrangementProps): JSX.Element => {
     };
   }
 
+  const captionsBannerProps = usePropsFor(_CaptionsBanner);
```
Comment (PorterNan) [inline packages/react-composites/src/composites/CallComposite/components/CallArrangement.tsx:159]: Let's move all hook calls earlier to the top of the component closure

File: packages/react-composites/src/composites/common/CaptionsBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+/** @private */
+export const CaptionsBanner = (): JSX.Element => {
+  const captionsBannerProps = usePropsFor(_CaptionsBanner);
+
+  return (
+    <>
+      {captionsBannerProps.captions.length > 0 && (
+        <Stack horizontalAlign="center">
+          <Stack.Item style={{ width: '50%' }}>
```
Comment (PorterNan) [inline packages/react-composites/src/composites/common/CaptionsBanner.tsx:17]: What does width:50% do here? it feels not that future proof to set width in a Stack.item (which is actually a flexbox), let's use something like flex-grow, flex-shrink or properties designed for flexbox

File: packages/react-composites/src/composites/CallComposite/components/CallArrangement.tsx [hunk 3]
```diff
@@ -185,6 +190,15 @@ export const CallArrangement = (props: CallArrangementProps): JSX.Element => {
                 />
               </Stack.Item>
             )}
+          <Stack.Item>
```
Comment (PorterNan) [inline packages/react-composites/src/composites/CallComposite/components/CallArrangement.tsx:193]: Use the common component instead

File: packages/react-composites/src/composites/CallComposite/components/CallArrangement.tsx (hunk 1 above)
Comment (carocao-msft) [reply packages/react-composites/src/composites/CallComposite/components/CallArrangement.tsx]: fixed

**Review by PorterNan — APPROVED** [review]

**Pull request merged**